packages/repo-tools/src/commands/openapi/lint.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Spectral,
+  Document,
+  Ruleset,
+  RulesetDefinition,
+} from '@stoplight/spectral-core';
+import { Yaml } from '@stoplight/spectral-parsers';
+import ruleset from '@apisyouwonthate/style-guide';
+import fs from 'fs-extra';
+import chalk from 'chalk';
+import { resolve } from 'path';
+import { runner } from './runner';
+import { YAML_SCHEMA_PATH } from './constants';
+import { oas } from '@stoplight/spectral-rulesets';
+import { DiagnosticSeverity } from '@stoplight/types';
+import { pretty } from '@stoplight/spectral-formatters';
+
+async function lint(
+  directoryPath: string,
+  config?: { skipMissingYamlFile: boolean; strict: boolean },
+) {
+  const { skipMissingYamlFile, strict } = config ?? {};
+  const openapiPath = resolve(directoryPath, YAML_SCHEMA_PATH);
+  if (!(await fs.pathExists(openapiPath))) {
+    if (skipMissingYamlFile) {
+      return;
+    }
+    throw new Error(`Could not find ${YAML_SCHEMA_PATH} in root of directory.`);
```

Comment:
might be useful to throw with `openapiPath` in there since it's not always clear what the root dir is